src/Microsoft.AspNet.Mvc.Core/DefaultControllerActionArgumentBinder.cs

```diff
@@ -66,6 +66,41 @@ public class DefaultControllerActionArgumentBinder : IControllerActionArgumentBi
             return actionArguments;
         }
 
+        public async Task<object> BindModelAsync(
+            ParameterDescriptor parameter,
+            ModelStateDictionary modelState,
+            OperationBindingContext operationContext)
+        {
+            var metadata = _modelMetadataProvider.GetMetadataForType(parameter.ParameterType);
+            var parameterType = parameter.ParameterType;
+            var modelBindingContext = GetModelBindingContext(
+                parameter.Name,
+                metadata,
+                parameter.BindingInfo,
+                modelState,
+                operationContext);
+
+            var modelBindingResult = await operationContext.ModelBinder.BindModelAsync(modelBindingContext);
+            if (modelBindingResult != null && modelBindingResult.IsModelSet)
+            {
+                var modelExplorer = new ModelExplorer(
+                    _modelMetadataProvider,
+                    metadata,
+                    modelBindingResult.Model);
+
+                var validationContext = new ModelValidationContext(
+                    modelBindingResult.Key,
+                    modelBindingContext.BindingSource,
+                    operationContext.ValidatorProvider,
+                    modelState,
+                    modelExplorer);
+                _validator.Validate(validationContext);
+                return modelBindingResult.Model;
+            }
+
+            return null;
```

Comment:
This is a good refactor. I had scoped out something similar thinking about building these tests.